lib/modules.nix

```diff
@@ -869,9 +869,17 @@ rec {
   # option declarations in submodules have accurate position information.
   # TODO: Merge this into mergeOptionDecls
   fixupOptionType = loc: opt:
-    if opt.type.getSubModules or null == null
+    builtins.addErrorContext ''
+      while evaluating the type of option `${showOption loc}`
```

Comment:
I don't think this is the right location for this, the type isn't actually evaluated here. Instead I'd put it near the `warnDeprecation` trigger, or maybe even in `mergeDefinitions` (maybe `warnDeprecation` should even be there too)